### PR TITLE
Respect scale when render bounding box around multiscale level

### LINKED ISSUE #6631: Transform mode problem when zoomed in on multiscale layer with `scale=` set
### 🐛 Bug Report

The transform box moves away from the image borders when zooming in on a multiscale image layer that is scaled (`scale=`).

This issue is very similar to https://github.com/napari/napari/issues/6382, which has been fixed by https://github.com/napari/napari/pull/6390.

Screenshot:

image

### 💡 Steps to Reproduce

Same code as in https://github.com/napari/napari/issues/6382, only differences:
- layer is scaled
- zoom in instead of out

```python
import napari
from skimage import data
from skimage.transform import pyramids

im = data.astronaut()[:,:,0].astype(float)

viewer = napari.Viewer()
layers = viewer.add_image([im] + [pyramids.pyramid_reduce(im, downscale=s) for s in [2]], scale=[10, 10])

viewer.layers.selection.active.mode = 'transform'

viewer.camera.zoom = 0.13
```

### 💡 Expected Behavior

That the transform box remains snapped to the image.

### 🌎 Environment

The problem occurs both on 

<details>
  <summary>main</summary>

napari: 0.4.19rc8
Platform: macOS-12.6-arm64-arm-64bit
System: MacOS 12.6
Python: 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:35:25) [Clang 16.0.6 ]
Qt: 5.15.8
PyQt5: 5.15.9
NumPy: 1.26.3
SciPy: 1.12.0
Dask: 2024.1.1
VisPy: 0.14.1
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.4

OpenGL:
  - GL version:  2.1 Metal - 76.3
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1440x900, scale 2.0
  - screen 2: resolution 1920x1080, scale 2.0
  - screen 3: resolution 1920x1080, scale 2.0

Settings path:
  - /Users/malbert/Library/Application Support/napari/napari-main_7341dd205874e3b46407ee91ab13d237573cf107/settings.yaml
Plugins:
  - napari: 0.4.19rc8 (77 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)

</details>

and

<details>
  <summary>0.4.19rc8</summary>

napari: 0.5.0a2.dev531+gdd993831
Platform: macOS-12.6-arm64-arm-64bit
System: MacOS 12.6
Python: 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:35:25) [Clang 16.0.6 ]
Qt: 5.15.8
PyQt5: 5.15.9
NumPy: 1.26.3
SciPy: 1.12.0
Dask: 2024.1.1
VisPy: 0.14.1
magicgui: 0.8.1
superqt: 0.6.1
in-n-out: 0.1.9
app-model: 0.2.4
npe2: 0.7.4

OpenGL:
  - GL version:  2.1 Metal - 76.3
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1440x900, scale 2.0
  - screen 2: resolution 1920x1080, scale 2.0
  - screen 3: resolution 1920x1080, scale 2.0

Settings path:
  - /Users/malbert/Library/Application Support/napari/napari-main_7341dd205874e3b46407ee91ab13d237573cf107/settings.yaml
Plugins:
  - napari: 0.5.0a2.dev531+gdd993831 (81 contributions)
  - napari-console: 0.0.9 (0 contributions)
  - napari-svg: 0.1.10 (2 contributions)

</details>

### 💡 Additional Context

It seems I'm just trying to break the transform mode in combination with multiscale and transforms 😅. The context is that I found the problem when looking into ways to have users manually edit transforms in [napari-stitcher](https://github.com/multiview-stitcher/napari-stitcher).

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

closes #6631

# Description

When calculating translate of bounding box for multiscale images, respect the layers scale

### CODE REVIEW

File: napari/_vispy/layers/base.py [hunk 1]
```diff
@@ -242,7 +242,7 @@ def _on_matrix_change(self):
                 ::-1
             ]
             - translate
-        )
+        ) / self.layer.scale[self.layer._slice_input.displayed][::-1]
```
Comment (brisvag) [inline napari/_vispy/layers/base.py]: suggestion: factor out `self.layer._slice_input.displayed[::-1]` to `displayed_vispy` or something like that, for readability.

Do you have a code snippet to test that this works and doesn't break?

Comment (Czaki) [reply napari/_vispy/layers/base.py]: The code snippet is in a linked issue.

I could make a followup PR with improved readability, but this change is done this way for easy cherry-pick.

**Comment by psobolewskiPhD** [comment]
I've pulled this down and the zoom issue (which is hard to spot) seems fixed.
The drag issue (on main try dragging the image to the left) is fixed.
Rotation and resize seems to work at both scale levels.

**Review by psobolewskiPhD — APPROVED** [review]

**Review by psobolewskiPhD — APPROVED** [review]
Grzegorz went over this with me today.
It makes sense and works, but it does mask some strange things with transforms.
But there's a caveat:
In the process of working through it, we found a bug with rotate transform.
The test image was this:
https://storage.googleapis.com/jax-public-ngff/data/0.4/Murray_Lab/ljb_20190114/Bach2_ERT2_e15_5%20_7_2.zarr/0/

(you will need napari-ome-zarr and I recommend NAPARI_ASYNC=1 it's amazing how well it works btw)
it's a multiscale 2D image, but has 3 channels and 2 singleton dims, so it loads as 3 layers and each is (1, 1, y, x)

With that image we noticed that if you rotate a layer and then translate the issue from the OP popped up again.
Further investigation indicated problems with the rotate matrix. Solving this would be quite involved, so here it's masked using the code in `if self.layer.affine.ndim > len(dims_displayed):`
Given that layer transforms and overlays only work in 2D this seems like a fine solution for the time being, but will require followup in the future.

**Pull request merged**